tests/test_utils.py

```
@@ -21,6 +23,24 @@
+def save_v8_coverage(cdp_session: Any, test_name: str) -> None:
+    """Collects V8 coverage data from CDP session and saves it directly in artifacts/."""
+    if not cdp_session:
+        return
+    try:
+        ARTIFACTS_DIR.mkdir(parents=True, exist_ok=True)
+        coverage = cdp_session.send("Profiler.takePreciseCoverage")
+        cdp_session.send("Profiler.stopPreciseCoverage")
+        output_file = ARTIFACTS_DIR / f"v8_coverage_{test_name}.json"
+        with open(output_file, "w", encoding="utf-8") as f:
+            json.dump(coverage, f)
+    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)